### DISCUSSION TITLE
Question about NUANMB editor

### DISCUSSION THREAD

**Discussion by Willy791** [ScanMountGoat/ssbh_editor#165] [Q&A]
Hello! I noticed you had options for editing NUANMB files that involve "Override Scale/Rotation/Translation" flags. Out of curiosity, what exactly do these options do? Do they simply just invalidate all changes made to that value, or does it still allow that group type to act with adjustments made to it?

For a more specific example, let's say I have a model with a bone at (0, 0, 500). I also have an animation that starts with a keyframe of (0, 0, 500) and ends at (0, 0, 1500). However, instead of using my bone's default position and simply moving it 1000 units in the Z direction like I want it to, the animation's value is added to the default position, resulting in the bone starting at (0, 0, 1000) and ending at (0, 0, 2000). Would enabling the "Override Translation" flag simply nullify my NUANMB or would it fix my issue entirely?

Thanks in advance for the response, this is a problem that's been haunting me for months

**Comment by ScanMountGoat** [comment]
The override flags disables the anim track's effect on the bone's translation, rotation, or scale. If you enabled "Override Translation" in your example, the bone would be locked to (0, 0, 500) for the duration of the animation. It's mainly used for special cases like thrown anims or swing bones where the transforms are set elsewhere.

> However, instead of using my bone's default position and simply moving it 1000 units in the Z direction like I want it to, the animation's value is added to the default position, resulting in the bone starting at (0, 0, 1000) and ending at (0, 0, 2000).

The animation basically "replaces" the skeleton transforms. A "T-Pose" animation doesn't have all 0's for translation, for example. Applying the animation on top of the base pose like you described would be "additive blending". I don't think additive blending is possible with anim files, but there are still some cases that haven't been tested yet. I would double check that you aren't also translating the bone's parent, which would combine the translations together. 

SSBH Editor and smash-ultimate-blender should give you pretty accurate previews for animations. Always do final testing in game of course. Please open an issue if you see any major differences between SSBH Editor and in game.
